test/server.test.js

```diff
@@ -2430,3 +2430,19 @@ test('should emit error with multiple next calls with strictNext', function(t) {
         });
     });
 });
+
+test('should have proxy event handlers as instance', function(t) {
+    var server = restify.createServer({
+        handleUpgrades: false
+    });
+    t.equal(server.proxyEvents.length, 7);
```

Comment:
I think I expect handleUpgrades: true to be 7, and 6 for false (may be related to the && indexOf check above)

Reply:
As per the doc

options.handleUpgrades Boolean Hook the upgrade event from the node HTTP server, pushing Connection: Upgrade requests through the regular request handling chain. (optional, default false) (http://restify.com/docs/server-api/). 

We are updating proxyEvents array when 
`options.handleUpgrades:false`
size will be 7 and in-case of true. we will skip it.

Reply:
Ah, I can't read. Must have been a long day. :D